libraries/tests/mbed/env/test_env.h

```diff
@@ -22,6 +22,36 @@ void notify_performance_coefficient(const char* measurement_name, const int valu
 void notify_performance_coefficient(const char* measurement_name, const unsigned int value);
 void notify_performance_coefficient(const char* measurement_name, const double value);
 
+// Host test auto-detection API
+void notify_host_test_name(const char *host_test);
+void notify_timeout(int timeout);
+void notify_test_id(const char *test_id);
+void notify_test_description(const char *description);
+
+// Host test auto-detection API
+#define TEST_START(TESTID)      notify_test_id(TESTID); notify_start()
```

Comment:
I would use prefix with the name of this test env, it's either part of mbed or any other. The other libraries can easily have similar macros TEST_DESCRIPTION or similar.